test/app-tap/datetime.test.lua

```diff
@@ -1163,20 +1163,52 @@ test:test("Time :set{} operations", function(test)
             'hour 6')
     test:is(tostring(ts:set{ min = 12, sec = 23 }), '2020-11-09T04:12:23+0300',
             'min 12, sec 23')
-    test:is(tostring(ts:set{ tzoffset = -8*60 }), '2020-11-08T17:12:23-0800',
+    test:is(tostring(ts:set{ tzoffset = -8*60 }), '2020-11-09T04:12:23-0800',
             'offset -0800' )
-    test:is(tostring(ts:set{ tzoffset = '+0800' }), '2020-11-09T09:12:23+0800',
+    test:is(tostring(ts:set{ tzoffset = '+0800' }), '2020-11-09T04:12:23+0800',
             'offset +0800' )
+    -- timestamp 1630359071.125 is 2021-08-30T21:31:11.125Z
     test:is(tostring(ts:set{ timestamp = 1630359071.125 }),
-            '2021-08-31T05:31:11.125+0800', 'timestamp 1630359071.125' )
-    test:is(tostring(ts:set{ msec = 123}), '2021-08-31T05:31:11.123+0800',
+            '2021-08-30T21:31:11.125+0800', 'timestamp 1630359071.125' )
+    test:is(tostring(ts:set{ msec = 123}), '2021-08-30T21:31:11.123+0800',
             'msec = 123')
-    test:is(tostring(ts:set{ usec = 123}), '2021-08-31T05:31:11.000123+0800',
+    test:is(tostring(ts:set{ usec = 123}), '2021-08-30T21:31:11.000123+0800',
             'usec = 123')
-    test:is(tostring(ts:set{ nsec = 123}), '2021-08-31T05:31:11.000000123+0800',
+    test:is(tostring(ts:set{ nsec = 123}), '2021-08-30T21:31:11.000000123+0800',
             'nsec = 123')
 end)
 
+test:test("Check :set{} and .new{} equal for all attributes", function(test)
+    test:plan(9)
+    local ts, ts2
+    local obj = {}
+    local attribs = {
+        {'year', 2000},
+        {'month', 11},
+        {'day', 30},
+        {'hour', 6},
+        {'min', 12},
+        {'sec', 23},
```

Comment:
Probably only `nsec` (because every next iteration this tests remember new attribute to check, and if `usec`/`msec` would be defined at the same time as `nsec` then we would get error. And I see no much point in overcomplicating this test to handle this corner case with `nsec`/`usec`/`msec`)